python/mxnet/optimizer.py

```diff
@@ -299,8 +300,13 @@ def update(self, index, weight, grad, state):
         if state:
             mom = state
             mom[:] *= self.momentum
-            mom[:] += -lr * (grad + wd * weight)
-            weight[:] += mom
+            grad += wd * weight
```

Comment:
I agree with @piiswrong that a subclassing would be better choice. and the update rule was not to long to write